lib/spack/external/archspec/json/cpu/microarchitectures.json

```diff
@@ -1044,7 +1044,6 @@
         "avx512vbmi",
         "avx512ifma",
         "sha_ni",
-        "umip",
```

Comment:
I think this is wrong, since potentially compilers can emit code accounting for this extension:
```
icelake-client
               Intel Icelake Client CPU with 64-bit extensions, MOVBE, MMX, SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2, POPCNT, PKU, AVX, AVX2, AES, PCLMUL, FSGSBASE, RDRND, FMA, BMI, BMI2, F16C, RDSEED, ADCX,
               PREFETCHW, CLFLUSHOPT, XSAVEC, XSAVES, AVX512F, AVX512VL, AVX512BW, AVX512DQ, AVX512CD, AVX512VBMI, AVX512IFMA, SHA, CLWB, **UMIP**, RDPID, GFNI, AVX512VBMI2, AVX512VPOPCNTDQ, AVX512BITALG,
               AVX512VNNI, VPCLMULQDQ, VAES instruction set support.
```
Reading [here](https://en.wikichip.org/wiki/x86/umip) it seems it's a security feature that can be switched on and off. I think our JSON file is correct. Hopefully your architecture was detected as `x86_64_v4` which means SIMD instructions are there?

Reply:
Right, UMIP was supposed to be a feature included in Cannon and Ice Lake. Perhaps its intentionally not mapped through our hypervisor? if that is the case, should we add another microarchitecture class for "icelake_aws" or something of the sort? 

```
ubuntu@ip-10-0-20-27:/shared/spack$ lscpu
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   46 bits physical, 48 bits virtual
CPU(s):                          8
On-line CPU(s) list:             0-7
Thread(s) per core:              2
Core(s) per socket:              4
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           106
Model name:                      Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz
Stepping:                        6
CPU MHz:                         3503.143
BogoMIPS:                        5799.99
Hypervisor vendor:               KVM
Virtualization type:             full
L1d cache:                       192 KiB
L1i cache:                       128 KiB
L2 cache:                        5 MiB
L3 cache:                        54 MiB
NUMA node0 CPU(s):               0-7
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves wbnoinvd ida arat avx512vbmi pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq rdpid md_clear flush_l1d arch_capabilities
```

Reply:
What is the output of:
```
$ spack arch
```
?

Reply:
without this fix: 
```
ubuntu@ip-10-0-20-27:/shared/spack$ spack arch
linux-ubuntu20.04-cascadelake
```

with it: 
```
ubuntu@ip-10-0-20-27:/shared/spack$ spack arch
linux-ubuntu20.04-icelake
```

Reply:
I checked GCC manual and assumptions on features being available foir a given microarchitecture, and what archspec does seems correct. Without `umip` the best match is `cascadelake`. Not sure what to do about this since we use names to reason about compatibility, so in some sense we also need compiler support for a given name. We had a similar case in https://github.com/archspec/archspec-json/issues/15 

Is naming causing any issue with respect to your use case? I.e. is there some feature that you need to use that is in `icelake` without `umip` and is not in `cascadelake`?